### DISCUSSION TITLE
Q: can we expose how much a rate limit has been exceeded?

### DISCUSSION THREAD

**Discussion by travispeloton** [vladimir-bukhtoyarov/bucket4j#408] [Q&A]
Hello!

I'm using `ConsumptionProbe.remainingTokens` to calculate when I'm within X percent of hitting the limit, so that is helping me monitor my rate limits.

For rate limits that I am exceeding, it would help me if I could see "by how much have we exceeded the limit?"

Can anyone suggest a way to expose this? Originally I thought that `remainingTokens` would turn negative, but it is locked to go not lower than `0` in the code

**Comment by vladimir-bukhtoyarov** [comment]
You can attach listener to the bucket and catch event each time when request to acquire tokens is being rejected https://bucket4j.com/8.4.0/toc.html#listening-for-bucket-events

**Reply by travispeloton** [reply]
I'm using the Redis implementation, and `bucket.tryConsumeAndReturnRemaining(1)`. I then have a counter for every time the bucket limit is exceeded, so I can look at that graph.

However, this graph has some unexpected results! Its not intuitive how much I need to increase the size of my bucket in some cases.

Anything wrong with how I am configuring it?

```
        BucketConfiguration.builder()
            .addLimit(Bandwidth.simple(1000, Duration.ofSeconds(5)))
            .build()
```

^ I have thought that this would be about 200 requests-per-second

**Reply by travispeloton** [reply]
My issues/confusion was because I didn't see that I have to do something to override old config in Redis.
